src/components/App.js

```diff
@@ -35,9 +37,19 @@ class Categories extends Component {
 
   renderCategories() {
     const { navigate } = this.props.navigation;
+    const { categories, isFetching } = this.props.categoryData;
+    if (isFetching) {
```

Comment:
I would personally move the loader out of the `renderCategories` and move into render() function, 

Something like 
```
{ isFetching && <Loading /> }
{ !isFetching && this.renderCategories() } 
```

You can define a functional component called Loading like this and just use it as above. 

```
const Loading = () => <Spinner
          type="FoldingCube"
           color="#3F51B5"
           size={100}
           style={{ marginTop: '50%' }}
         />;
```


If you'd like to keep the code even cleaner, create a new component called Categories (a different file) and just use <Categories data={categories} /> and move the rendering logic into its render method.

You can extract and expose the required props from `mapStateToProps`. Would make things look cleaner.
```
const mapStateToProps = ({ data: { categories, isFetching }) => ({
  categories,
  isFetching,
});
```